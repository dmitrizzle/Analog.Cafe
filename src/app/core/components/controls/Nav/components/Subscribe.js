import styled from "styled-components"
export default styled.div`
  width: 10em;
  position: absolute;
  border-radius: 0.33em;
  box-shadow: 0 1px ${props => props.theme.color.foreground()};
  color: ${props => props.theme.color.brand()};
  height: 0.65em;
  top: 5em;
  margin-left: calc(50vw - 20em);
  z-index: 1;
  @media (max-width: 48em) {
    display: none;
  }
  a {
    font-style: italic;
    font-family: "Lora", Georgia, serif;
    display: block;
    background: ${props => props.theme.color.background()};
    width: 6.5em;
    text-align: center;
    margin: 0 auto -1em;
    height: 1.25em;
    text-decoration: none;
  }
`
