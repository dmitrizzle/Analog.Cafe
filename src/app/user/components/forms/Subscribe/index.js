import React from "react"

import { GA } from "../../../../utils"
import { TEXT_LABELS } from "../../../../core/constants/messages-"
import CardButton from "../../../../core/components/controls/Card/components/CardButton"
import MailChimpPrefill from "./components/MailChimpPrefill"

export default class extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      subscribeForm: false
    }
    this.handleRevealSubscribeForm = this.handleRevealSubscribeForm.bind(this)
  }
  handleRevealSubscribeForm = event => {
    event.preventDefault()
    event.stopPropagation()
    this.setState({
      subscribeForm: !this.state.subscribeForm
    })
    this.props.subscribeFormCallback &&
      this.props.subscribeFormCallback(this.state.subscribeForm)
    GA.event({
      category: "Campaign",
      action: "ActionsCard.quickSubscribe_open",
      label: this.props.formLocation ? this.props.formLocation : null
    })
  }
  handleSubmitCallback = () => {
    this.setState({
      subscribeForm: !this.state.subscribeForm
    })
  }
  componentWillReceiveProps = nextProps => {
    if (
      nextProps.stateOverwrite === null ||
      nextProps.stateOverwrite === undefined
    )
      return
    this.setState({
      subscribeForm: nextProps.stateOverwrite
    })
  }
  render = () => {
    return (
      <div>
        {!this.state.subscribeForm ? (
          <CardButton branded onClick={this.handleRevealSubscribeForm}>
            {TEXT_LABELS.SUBSCRIBE}
          </CardButton>
        ) : (
          <MailChimpPrefill
            formLocation={this.props.formLocation}
            buttonText={TEXT_LABELS.SUBMIT}
            autoFocus
            submitCallback={this.handleSubmitCallback}
          />
        )}
      </div>
    )
  }
}
