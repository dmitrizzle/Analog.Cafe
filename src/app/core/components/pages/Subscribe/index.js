import React from "react"
import styled from "styled-components"

import { CardCaptionIntegrated } from "../../controls/ArticleActions/components/Options"
import { makeFroth } from "../../../../utils"
import ArticleSection from "../Article/components/ArticleSection"
import ArticleWrapper from "../Article/components/ArticleWrapper"
import ButtonGroup from "../../controls/Button/components/ButtonGroup"
import CardHeader from "../../controls/Card/components/CardHeader"
import CardIntegrated from "../../controls/Card/components/CardIntegrated"
import HeaderLarge from "../../vignettes/HeaderLarge"
import Link from "../../controls/Link"
import MetaTags from "../../vignettes/MetaTags"
import Subscribe from "../../../../user/components/forms/Subscribe"

const metaTitle = "Analogue Reads"
const metaDescription =
  "Photo essays, reviews, guides. Weekly email newsletter, delivered Tuesdays at 9AM EST."

const bgList = [
  "image-froth_1480456_Hk9sPRsQV",
  "image-froth_1469613_Skk4VZmZE",
  "image-froth_1518843_B1skGpWWV",
  "image-froth_1508069_r1vQk9iJN",
  "image-froth_1206996_r1CqlUwRm",
  "image-froth_1491916_rk1gPcNtm",
  "image-froth_1557196_SJmddu5y7",
  "image-froth_1500270_ryAMqwpRz"
]
const bgRoulette = () => bgList[Math.floor(Math.random() * bgList.length)]
const bgRouletteCached = bgList[Math.floor(Math.random() * bgList.length)]

export const WallPaper = styled.div`
  border-top: ${props => props.theme.elements.thickBorder};
  margin-top: 1em;
  height: 100vh;
  min-height: 670px;
  padding-top: 3em;
  background: url(${props =>
    makeFroth({
      src: props.bgRouletteCached,
      size: "l"
    }).src});
  background-size: cover;
  background-position: top center;
  ${props => props.theme.size.breakpoint.max.m`
    margin-top: 0;
    background-image: url(${props =>
      makeFroth({
        src: props.bgRouletteCached,
        size: "m"
      }).src});
  `};
`

export const GetYourWeekly = () => (
  <CardCaptionIntegrated>
    New photo essays, reviews, and guides every Tuesday at 9AM EST.{" "}
    <Link to="/privacy-policy">No spam</Link>.
    {/* {" "}
    <Link to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=f8892b3a23">
      Sneak Peek
    </Link>
    . */}
  </CardCaptionIntegrated>
)

export const SubscribeWrapper = styled(ArticleWrapper)`
  overflow: visible;
  padding-top: ${props => (props.embed ? "6em" : undefined)};
  @media print {
    display: none;
  }
`

export default props => {
  return (
    <SubscribeWrapper>
      {!props.embed && (
        <React.Fragment>
          <MetaTags metaTitle={metaTitle} metaDescription={metaDescription} />
          <HeaderLarge
            pageTitle={metaTitle}
            pageSubtitle="Every Tuesday Morning"
          />
        </React.Fragment>
      )}

      <WallPaper
        bgRouletteCached={bgRouletteCached}
        bgRoulette={!props.cached && bgRoulette}
      >
        {props.embed && (
          <HeaderLarge
            pageTitle={metaTitle}
            pageSubtitle="Every Tuesday Morning"
            style={{ textSshadow: "0 0 5em rgba(255, 255, 255, 0.64)" }}
          />
        )}
        <ArticleSection>
          <ButtonGroup>
            <CardIntegrated>
              <CardHeader
                stubborn
                buttons={[0]}
                noStar
                title="ðŸ’Œ Weekly Email Newsletter"
              />
              <GetYourWeekly />
              <Subscribe
                autoFocus={!props.embed}
                stateOverwrite={{ subscribeForm: true }}
                formLocation={"Subscribe"}
              />
            </CardIntegrated>
            {/* <FollowButtons /> */}
          </ButtonGroup>
        </ArticleSection>
      </WallPaper>
    </SubscribeWrapper>
  )
}
