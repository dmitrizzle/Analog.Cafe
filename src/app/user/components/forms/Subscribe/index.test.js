import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import Subscribe from "./"

it("Render Subscribe, matches snapshot", () => {
  const element = shallow(<Subscribe />)
  expect(element).toMatchSnapshot()
})
