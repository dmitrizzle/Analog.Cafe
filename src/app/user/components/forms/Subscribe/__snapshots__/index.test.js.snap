// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Subscribe, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Styled(Component)
        branded={true}
        onClick={[Function]}
>
        Email Newsletter ❤︎
</Styled(Component)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "branded": true,
        "children": "Email Newsletter ❤︎",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Email Newsletter ❤︎",
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Styled(Component)
          branded={true}
          onClick={[Function]}
>
          Email Newsletter ❤︎
</Styled(Component)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "branded": true,
          "children": "Email Newsletter ❤︎",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Email Newsletter ❤︎",
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
